data/lib/src/datasource_impl/received_share_datasource_imp.dart

```diff
@@ -91,4 +94,27 @@ class ReceivedShareDataSourceImp extends ReceivedShareDataSource {
 
     return taskIds.map((taskId) => DownloadTaskId(taskId)).toList();
   }
+
+  @override
+  Future<Uri> downloadPreviewReceivedShare(ReceivedShare receivedShare,
+      DownloadPreviewType downloadPreviewType, Token permanentToken, Uri baseUrl, CancelToken cancelToken) {
+    var downloadUrl;
+    if (downloadPreviewType == DownloadPreviewType.original) {
+      downloadUrl = Endpoint.receivedShares
+          .downloadServicePath(receivedShare.shareId.uuid)
+          .generateDownloadUrl(baseUrl);
+    } else {
+      downloadUrl = Endpoint.receivedShares
```

Comment:
oh, have you tested the download url for this case ?

Reply:
Yes, i tested it